src/pages/iou/MoneyRequestDescriptionPage.js

```diff
@@ -34,14 +44,36 @@ class MoneyRequestDescriptionPage extends Component {
         super(props);
 
         this.updateComment = this.updateComment.bind(this);
+        this.navigateBack = this.navigateBack.bind(this);
+        this.iouType = lodashGet(props.route, 'params.iouType', '');
+        this.reportID = lodashGet(props.route, 'params.reportID', '');
     }
 
-    /**
-     * Goes back and clears the description from Onyx.
-     */
-    onBackButtonPress() {
-        IOU.setMoneyRequestDescription('');
-        Navigation.goBack();
+    componentDidMount() {
+        const moneyRequestId = `${this.iouType}${this.reportID}`;
+        const shouldReset = this.props.iou.id !== moneyRequestId;
+        if (shouldReset) {
+            IOU.resetMoneyRequestInfo(moneyRequestId);
+        }
+
+        if (_.isEmpty(this.props.iou.participants) || this.props.iou.amount === 0 || shouldReset) {
+            Navigation.goBack(ROUTES.getMoneyRequestRoute(this.iouType, this.reportID), true);
+        }
+    }
+
+    // eslint-disable-next-line rulesdir/prefer-early-return
+    componentDidUpdate(prevProps) {
+        // ID in Onyx could change by initiating a new request in a separate browser tab or completing a request
+        if (_.isEmpty(this.props.iou.participants) || this.props.iou.amount === 0 || prevProps.iou.id !== this.props.iou.id) {
+            // The ID is cleared on completing a request. In that case, we will do nothing.
+            if (this.props.iou.id) {
+                Navigation.goBack(ROUTES.getMoneyRequestRoute(this.iouType, this.reportID), true);
+            }
+        }
+    }
```

Comment:
This code need to be documented better. It is not obvious at all what is happening here. Please explain in the code what "reset" means in this context and what an `iou.id` even is? When did we start adding "id" to the IOU and why...

Reply:
We start to add an ID to avoid a problem explained here https://github.com/Expensify/App/pull/17964#issuecomment-1584422120. The ID is iouType + reportID
https://github.com/Expensify/App/blob/e6c4b4e2f0bf9f42e57bebdd7720ee776ab6c93a/src/pages/iou/MoneyRequestDescriptionPage.js#L53-L54

Every time we start a money request, it will set an ID to it
https://github.com/Expensify/App/blob/e6c4b4e2f0bf9f42e57bebdd7720ee776ab6c93a/src/libs/actions/IOU.js#L1298-L1301

> what "reset" means

I think this can be found in the function itself.
https://github.com/Expensify/App/blob/e6c4b4e2f0bf9f42e57bebdd7720ee776ab6c93a/src/libs/actions/IOU.js#L65-L69

Inspired from
https://github.com/Expensify/App/blob/e6c4b4e2f0bf9f42e57bebdd7720ee776ab6c93a/src/libs/actions/Task.js#L16-L21

Reply:
I would appreciate it if the code could be annotated more thoroughly. Enhancing code documentation is essential for our team's effectiveness. Can we kindly address this?

Regarding the current solution, I remain uncertain about its efficiency. It appears more like a workaround to me, rather than a refined resolution.

Reply:
Sure, I can create a follow-up PR to add the comment and fix this typo https://github.com/Expensify/App/pull/17964#discussion_r1236812199

But, should we wait until the regression period ends before opening the PR?

> Regarding the current solution, I remain uncertain about its efficiency. It appears more like a workaround to me, rather than a refined resolution.

I can't say for sure whether it's a workaround or not, but at least that's what we can come up with considering so many [cases](https://github.com/Expensify/App/pull/17964#issuecomment-1588849643) using Onyx.